### DISCUSSION TITLE
请教一下ListBox ItemTemplateSelector的更新问题

### DISCUSSION THREAD

**Discussion by GSCmax** [BYJRK/DotNet-Discussions#75] [MVVM]
各位大哥请教一下，WPF ListBox的ItemTemplateSelector在程序载入的时候可以根据元素属性成功选择模板，但是元素属性变化的时候模板不会自动更新，请问需要如何修改代码呢。后台数据已经使用ObservableProperty标记并且更改可以展示到前台，但是模板不会自动更新。

**Comment by xzf888** [comment]
变量类型是 ObservableCollection   吗?

**Reply by GSCmax** [reply]
是的

**Comment by CodingOctocat** [comment]
代码贴出来。

**Reply by GSCmax** [reply]
已经弃用原先的逻辑了，绕道了绕道了
